torch/_inductor/spmd_prefetch.py

```diff
@@ -0,0 +1,128 @@
+# mypy: allow-untyped-defs
+# pyre-strict
+import typing
+from typing import (
+    Any,
+    Counter,
+    DefaultDict,
+    Dict,
+    Generic,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+from . import config, ir, comms, scheduler
+from collections import OrderedDict
+
+import torch
+
+from .dependencies import WeakDep
+from .utils import is_collective, is_wait, tuple_sorted
+
+
+    
+def reorder_all_gather(
+    snodes: List["scheduler.BaseSchedulerNode"],
+    all_gather_order: Optional[str] = "before",
```

Comment:
Since this is a binary option, it might be better to use a boolean variable to denote.
E.g. call it `all_gather_before_last_wait = True`